stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/MessagePreviewFormatter.kt

```diff
@@ -97,6 +99,7 @@ private class DefaultMessagePreviewFormatter(
     /**
      * Appends the message text to the [AnnotatedString].
      */
+    @SuppressLint("ComposableNaming")
     @Composable
```

Comment:
I think we should try and remove the Composable annotation from these functions as for the most part, these do not rely on Compose at all.

The only Compose requirement here are the fontStyles and the AttachmentFactories, which we can pass to the messageFormatter as parameters.

Here's an example of what it would look like for `appendAttachmentText()`:

```
    /**
     * Appends a string representations of [attachments] to the [AnnotatedString].
     */
    private fun AnnotatedString.Builder.appendAttachmentText(
        attachments: List<Attachment>,
        attachmentFactories: List<AttachmentFactory>,
        attachmentTextFontStyle: FontStyle,
    ) {
        if (attachments.isNotEmpty()) {
            attachmentFactories
                .firstOrNull { it.canHandle(attachments) }
                ?.textFormatter
                ?.let { textFormatter ->
                    attachments.mapNotNull { attachment ->
                        textFormatter.invoke(attachment)
                            .let { previewText ->
                                previewText.ifEmpty { null }
                            }
                    }.joinToString()
                }?.let { attachmentText ->
                    val startIndex = this.length
                    append(attachmentText)

                    addStyle(
                        SpanStyle(
                            fontStyle = attachmentTextFontStyle
                        ),
                        start = startIndex,
                        end = startIndex + attachmentText.length
                    )
                }
        }
    }
```

Notice how with just a couple of parameters we managed to make this a regular function. The base function signature would be the following:

```
    public fun formatMessagePreview(
        message: Message,
        currentUser: User?,
        textFontStyle: FontStyle,
        senderNameFontStyle: FontStyle,
        attachmentTextFontStyle: FontStyle,
        attachmentFactories: List<AttachmentFactory>,
    ): AnnotatedString
```
Which makes sense, given that this is an implementation detail of our SDK and if our users want to override it or display something completely different, they probably won't use our `ChannelDetails`.

When we add the Composable annotation, Compose adds a new entry to its tree that represents the value of that Composable - in this case a UI-less, value-less function, that appends some text to a builder.

Since these Composables produce absolutely no observable result -we don't return a value that's powered by a `remember` block, we don't produce UI with this, we don't rely on anything that's likely to change in the runtime (factories and typography don't change often), we're adding unnecessary entries to the tree that might cause performance overhead.

I'm not 100% sure of the impact of this, but it's also easier to have the SAM/functional interface approach, if the functions are not Composables.

LMK what you think - I can make the proposed changes if you want, as I have them stashed locally.